application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,319 @@
+
+        getData().addSubcommands(allRoles, selectRoles);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+        if (!member.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("You dont have the right permissions to use this command")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        Member selfMember = Objects.requireNonNull(event.getGuild()).getSelfMember();
+        if (!selfMember.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("The bot needs the manage role permissions").setEphemeral(true).queue();
+            logger.warn("The bot needs the manage role permissions");
+            return;
+        }
+
+        SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+        boolean ephemeral = false;
+
+        if (Objects.equals(event.getSubcommandName(), CHOOSE_OPTION)) {
+            addMenuOptions(event, menu, "Select the roles to display", 1);
+            ephemeral = true;
+        } else {
+            addMenuOptions(event, menu, "Select your roles", 0);
+        }
+
+        // Handle Optional arguments
+        OptionMapping titleOption = event.getOption(TITLE_OPTION);
+        OptionMapping descriptionOption = event.getOption(DESCRIPTION_OPTION);
+
+        String title = handleOption(titleOption);
+        String description = handleOption(descriptionOption);
+
+        MessageBuilder messageBuilder = new MessageBuilder(makeEmbed(title, description))
+            .setActionRows(ActionRow.of(menu.build()));
+
+        if (ephemeral) {
+            event.reply(messageBuilder.build()).setEphemeral(true).queue();
+        } else {
+            event.getChannel().sendMessage(messageBuilder.build()).queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+        }
+    }
+
+    /**
+     * Adds role options to a selection menu
+     * <p>
+     *
+     * @param event the {@link SlashCommandEvent}
+     * @param menu the menu to add options to {@link SelectionMenu.Builder}
+     * @param placeHolder the placeholder for the menu {@link String}
+     * @param minValues the minimum number of selections. nullable {@link Integer}
+     */
+    private static void addMenuOptions(@NotNull final Interaction event,
+            @NotNull final SelectionMenu.Builder menu, @NotNull final String placeHolder,
+            @Nullable final Integer minValues) {
+
+
+        Guild guild = Objects.requireNonNull(event.getGuild(), "The given guild cannot be null");
+
+        Role highestBotRole = guild.getSelfMember().getRoles().get(0);
+        List<Role> guildRoles = guild.getRoles();
+
+        Collection<Role> roles = new ArrayList<>(
+                guildRoles.subList(guildRoles.indexOf(highestBotRole) + 1, guildRoles.size()));
```

Comment:
This logic might be weird to someone not familiar with it, I would recommend moving it into a helper method like `getRolesBotCanUse` or similar (maybe u have a better name)